README.md

```diff
@@ -0,0 +1,150 @@
+# NimbleHex
+
+NimbleHex allows self-hosting of Hex packages.
+
+Features:
+
+* Pluggable storage. NimbleHex ships with local filesystem and S3 implementations
+* Publishing packages via HTTP API
+* Support for multiple repositories
+* Mirroring
+
+## Setup
```

Comment:
I think it's important for this project to ship with step-by-step instructions both on setting up the server as well as configuring clients to use it. I intend to also write a similar guide to use with Rebar.

I wonder if README is the best place for such "guides", perhaps we should turn these into ExDoc pages and publish on hex docs? (without actually publishing nimble_hex as package). Or add these instructions to https://hex.pm/docs/usage?

I think we should iterate on docs in the README, but just something to think about for the future.

Reply:
> or are you suggesting pushing nimble_hex as a package?

Initially I was considering this but for now decided against as it's adding complexity that we don't need (if we do that we'd likely make dependency on ex_aws optional, etc). Yeah, I was thinking about publishing just to hex docs but not publishing it as a package. Agreed, let's keep it simple for now. USAGE.md seems like a good idea to, will look into it.